dex/src/main/scala/com/wavesplatform/dex/actors/address/AddressActor.scala

```diff
@@ -71,7 +73,53 @@ class AddressActor(owner: Address,
   private var wsAddressState              = WsAddressState.empty(owner)
   private var wsSendSchedule: Cancellable = Cancellable.alreadyCancelled
 
-  override def receive: Receive = {
+  // if (started) because we haven't pendingCommands during the start
+  private val eventsProcessing: Receive = {
+    case event: OrderAdded =>
+      import event.order
+      log.debug(s"OrderAdded(${order.id}, ${event.reason}, ${event.timestamp})")
+      refreshOrderState(order, event)
+      if (started) pendingCommands.remove(order.id).foreach { command =>
+        log.trace(s"Confirming placement for ${order.id}")
+        command.client ! Event.OrderAccepted(order.order)
+      }
+
+    case event: OrderExecuted =>
+      log.debug(s"OrderExecuted(${event.submittedRemaining.id}, ${event.counterRemaining.id}), amount=${event.executedAmount}")
+      List(event.submittedRemaining, event.counterRemaining).filter(_.order.sender.toAddress == owner).foreach(refreshOrderState(_, event))
+      if (started) context.system.eventStream.publish(CreateExchangeTransactionActor.OrderExecutedObserved(owner, event))
+
+    case event @ OrderCanceled(ao, reason, ts) =>
+      val id       = ao.id
+      val isActive = activeOrders.contains(id)
+      log.debug(s"OrderCanceled($id, $reason, $ts, isActive=$isActive)")
+      if (isActive) refreshOrderState(ao, event)
+      if (started) pendingCommands.remove(id).foreach { pc =>
+        log.trace(s"Confirming cancellation for $id")
+        pc.client ! Event.OrderCanceled(id)
+      }
+
+    case command @ OrderCancelFailed(id, reason) =>
+      if (started) pendingCommands.remove(id) match {
+        case None => // Ok on secondary matcher
+        case Some(pc) =>
+          log.trace(s"Got $command, sending a response to a client")
+          pc.client ! reason
+      }
+  }
+
+  private val failuresProcessing: Receive = { case Status.Failure(e) =>
+    log.error(s"Got $e", e)
+  }
+
+  private val starting: Receive = eventsProcessing orElse failuresProcessing orElse {
+    case AddressDirectoryActor.StartWork =>
+      activeOrders.values.foreach(x => scheduleExpiration(x.order))
+      started = true
+      context.become(working)
+  }
+
+  private val working: Receive = eventsProcessing orElse failuresProcessing orElse {
```

Comment:
Cases below can be grouped in `commandProcessing` metod

Reply:
Will do this next time